### PR TITLE
Allow loading documents by relationship and externalId

### PR DESCRIPTION
**Pull request by ivan-bn**
Documents can now be loaded using a _relationships_ or _externalIds_ value.

### CODE REVIEW

File: test/feature/entity-feature.js [hunk 1]
```diff
@@ -13,6 +13,22 @@ Feature("Entity", () => {
...
+        "id": "type.guid"
+      },
+      {
+        "system": "other.system.name",
+        "type": "other.foreign.type",
+        "id": "other.type.guid"
+      }
+    ],
+    externalIds: {
+      "externalIdType": "external.id",
```
Comment (markusn) [inline test/feature/entity-feature.js]: Ett alternativ vore:
`externalIds: { "system": {"type": "externalId"}, "other.system": {"other.type": "otherExternalId"}}} `

File: lib/crud.js [hunk 2]
```diff
@@ -33,6 +33,40 @@ function load(id, force, cb) {
   });
 }
 
+function loadByRelationship(entityType, relationType, id, cb) {
```
[the remark this answers is not in the document]
Comment (ivan-bn) [reply lib/crud.js]: Kastar fel vid `res.rows.length > 1` nu.

File: index.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 "use strict";
 
 const pgClient = require("./lib/client");
-const crud = require("./lib/crud");
+const crud = require("./lib/query");
```
Comment (jens-peterolsson) [inline index.js]: *query* = require("./lib/query");

Comment (jens-peterolsson) [reply index.js]: Också exportera som query, inte som crud

**Pull request merged**